### ISSUE TITLE
VK_ERROR_DEVICE_LOST

### ISSUE THREAD

**Issue by MrChebik** [doitsujin/dxvk#469]
I've already met this problem `VK_ERROR_DEVICE_LOST` in two games:

- **Batman: Arkham Origins**
- **Fallout 4**

At the **Batman: Arkham Origins**, I get this during the game launch, seeing the game logo, and after the black screen.
At the **Fallout 4**, I get this in time for loading a new game.

### Software information
All settings in both games to low, resolution to 864x486.
**Batman: Arkham Origins**:
- Tried with window.
- Tried without Vsync.

### System information
- GPU: Intel HD 5500
- Driver: Mesa 18.1.2
- Wine version: 3.10-staging
- DXVK version: 0.61

### Log files
**Batman: Arkham Origins**:
- [apitrace](https://drive.google.com/file/d/1iwZDcyH2Fjc17qrWexJd2jiJ54W4ngJB/view?usp=sharing) x32
- [d3d11.log](https://pastebin.com/qwdg4jnS)
- [dxgi.log](https://pastebin.com/njhmLMDx)

**Fallout 4**
- [apitrace](https://drive.google.com/file/d/1wVaQZl-LQ7jeZbyf_GkYD73XGXp1tbAl/view?usp=sharing) x64
- [d3d11.log](https://pastebin.com/QWkHMVBq)
- [dxgi.log](https://pastebin.com/5UwR6nyQ)

**Comment by doitsujin** [comment]
Can't reproduce, please run the games with Vulkan validation layers enabled and post the full output.

**Comment by MrChebik** [comment]
This is strange, but **Fallout 4** does not display as much information as **Batman: Arkham Origins**. [Google Drive](https://drive.google.com/open?id=121gtvZI18BitHOEQY2gmrHKlL5cxLJGe)

**Comment by doitsujin** [comment]
And the errors in Batman all occur **after** the device lost error, making them effectively meaningless. Can't really do anything.

That said, does your Broadwell GPU have *full* Vulkan support? I know for a fact that Haswell doesn't. It'd be great to know if this works on Skylake or not.

**Comment by MrChebik** [comment]
I can not say for sure, but I do not have any problems with **Dota 2**, **The Talos Principle** on Vulkan.

**Comment by MrChebik** [comment]
**Prey - Mooncrash** - Also not worked.
I can start playing. A load of equipment arrives to me, I insert it for virtual reality, after I sit down on an armchair. There is a load of people to save, and hangs.

[Google Drive](https://drive.google.com/file/d/1894F3qWSgFelyoqiaqlKQ8dlwq0UHwcE/view?usp=sharing)

**Comment by MrChebik** [comment]
I think, I did not correct write **apitrace**, I'll overwrite them.

**Comment by MrChebik** [comment]
The new game:

- Outlast 2

At the start game, often hangs after listening to a fragment of the memory of a girl.

**Comment by jekstrand** [comment]
Yes, Broadwell is fully Vulkan conformant and stuff should work there.  Before filing any bugs against mesa, though, please run with the validation layers and ensure that validation is happy.  If it is, please file a mesa bug and we'll look into it.  Broadwell is, however, somewhat more limted than Sky Lake so it could be that DXVK is hitting unexpected limitations.

**Comment by doitsujin** [comment]
If this is true, this is devastating news. Is there *any* way at all to narrow down what this might be?

I double-checked the code and did end up finding a small resource tracking issue related to framebuffer attachments (fixed in 67fe452580222fc082418ee0bc6d91dfd2f825e2), but that one would only cause gpu->cpu synchronization to fail when the game tries to map a render target.

**Comment by jekstrand** [comment]
Yeah, it would be very bad.  Unfortunately, I wasn't able to narrow it down any further today. Tomorrow, I'll throw some more instrumentation in the driver and see if I can find the busy buffer.

**Comment by jekstrand** [comment]
I'm also somewhat suspicious of that conclusion because Skyrim is rock-solid on Sky Lake and a busy tracking problem doesn't seem likely to only affect one platform.  I'll keep digging.

**Comment by jekstrand** [comment]
Quick update:  Things definitely seem harder to hang today since upgrading DXVK so 67fe452 may have helped.

**Comment by jekstrand** [comment]
Update:  I think I finally have at least the Skyrim hang narrowed down.  It's a very Broadwell-specific issue and I think I've got a fix.  Haven't tested any of the others yet though.

**Comment by MrChebik** [comment]
@jekstrand I tried your **anv-dxvk** branch. At the **Skyrim SE** I got to the Whiterun, everything went smoothly, thank you!

As for other games - it seems the problem remains. **Fallout 4** - did not work out.
Although the error **VK_ERROR_DEVICE_LOST** does not appear there explicitly.

To give a more accurate answer, I need to run a **Batman: Arkham Xxx**, but I do not know how to build a 32-bit version of Mesa. The default is 64-bit.

--
Also I will try **Prey: Mooncrash** in a short time.

**Comment by jekstrand** [comment]
I got Batman: Arkham Origins to run by which I mean that I got it to hit an assertion in my debug build of mesa.  Definitely a driver bug. :smiley:

**Comment by jekstrand** [comment]
The batman hang was caused by us not properly implementing `vkCmdClearColorImage` on linear RGB images.  Patches to fix it are in my wip/anv-dxvk branch.
